sys/kern/subr_devmap.c

```diff
@@ -223,6 +223,14 @@ devmap_ptov(vm_paddr_t pa, vm_size_t size)
 	for (pd = devmap_table; pd->pd_size != 0; ++pd) {
 		if (pa >= pd->pd_pa && pa + size <= pd->pd_pa + pd->pd_size)
 #ifdef __CHERI_PURE_CAPABILITY__
+			/*
+			 * There is no way to ensure exact representablity here
+			 * because the devmap table may be loaded from FDT or
+			 * some other means that is oblivious to CHERI.
+			 * XXX-AM:
+			 * Perhaps we should try to be exact or add an
+			 * assertion to fail explicitly?
+			 */
```

Comment:
I wonder if we should have the ability to warn when the mapping is inexact?

Reply:
It might be interesting to teach dtc complain about unrepresentable regions and do a scan of the linux FDT collection.

Reply:
I think FDT code is the only place that uses this in a way that might be unrepresentable, perhaps we can make this `setboundsexact()` and complain when loading the static device addresses if they end up not being representable + teach dtc about it.

Reply:
I did that as part of the precision analysis and I believe it was all fine but we should double check